src/serde/ser.rs

```diff
@@ -13,14 +13,14 @@ pub struct Serializer {
     state: State,
 }
 
-// NBT has a different order of things, then most other formats
-// So I use State, to keep what serializer has to do, and some information like field name
+// NBT has a different order of things, than most other formats,
+// So I use the `State` struct to keep what serializer has to do and some information like field name
```

Comment:
maybe without the I? feels weird to see I in code/docs
```suggestion
// So the `State` struct is used to keep what serializer has to do, and some information like field name
```